website/content/en/docs/best-practices/observability-best-practices.md

```diff
@@ -0,0 +1,268 @@
+    2.1  Alerts should be implemented based on the guidelines below.  
+    2.2. **Alerts Runbooks** - Each alert can include a `runbook_url` annotation and an alert runbook that describes it. See additional details below.  
+    2.3. **Alerts Tests** - Alerts should include E2E Testing and unit tests.  
+3. **Events** - Custom Resources should emit custom events for the operations taking place.
+
+Additional component would be `Dashboards` and `Logs`, which are not addressed in this document at this point.  
+
+### Operators Observability General Guidelines
+**Important:** It is highly recommended to separate your monitoring code from your core operator code.  
+We recommend to create a dedicated `/monitoring` subfolder that will include all the code of the [Operator Observability Recommended Components](#operator-observability-recommended-components), that are outlined above. For example, in the [memcached-operator](https://github.com/operator-framework/operator-sdk/tree/master/testdata/go/v4-alpha/monitoring/memcached-operator/monitoring).
+
+In your core operator code only call the functions the functions that will update the metrics value from you desired location. For example, in the [memcached-operator](https://github.com/operator-framework/operator-sdk/blob/367bd3597c30607099aa73637f5286f7120b847a/testdata/go/v3/monitoring/memcached-operator/controllers/memcached_controller.go#L242).
+
+All operators start small.This separation will help you, as a developer, with easier maintenance of both your operator core code and the monitoring code and for other stakeholders to understand your monitoring code better.
+
+### Metrics Guidelines
+
+#### Metrics Naming
+Note: The metrics naming is based on the [OpenMetrics](https://github.com/OpenObservability/OpenMetrics) and [Prometheus exposition formats](https://prometheus.io/docs/instrumenting/exposition_formats/) and adds to it.  
+
+Your operator metrics should align with the Kubernetes metrics names.
+
+Your operator users should get the same experience when searching for a metric across Kubernetes operators, resources and custom resources.
+1. Check if a similar Kubernetes metric, for node, container or pod, exists and try to align to it.
+2. The metrics search list in the Prometheus and Grafana UI is sorted in alphabetical order.
+When searching for a metric, it should be easy to identify metrics that are related to a specific operator.
+That is why we recommend that your operator metrics name to follow this format:
+`Operator name` prefix + the `Sub operator name` or `entity` + `metric name` based on the [Prometheus naming conventions](https://prometheus.io/docs/practices/naming/). For example, in the [memcached-operator](https://github.com/operator-framework/operator-sdk/blob/0d2fa86f0d3cc92c4672cb9e1d246efaefcf7ced/testdata/go/v4-alpha/monitoring/memcached-operator/monitoring/metrics.go#L14).
+
+**Note:** In Kubernetes metrics are separated like this:
```

Comment:
Can we have a link of the reference to those `kuberntes metrics`?

Reply:
Its actually the Prometheus node exporter metrics. I updated the text and added the link.